<launch>
    <arg name="sampling_freq" default="1e6"/>
    <arg name="pinger_freq" default="27e3"/>
    <arg name="noise_stdev" default="10"/>
    <arg name="x" default="23.9"/>
    <arg name="y" default="-23.6" />
    <arg name="rate" default="1" />

    <!--simulator-->
    <node
        pkg="au_sonar"
        type="sonar_sim.py"
        name="sonar_sim"
        output="screen"
        respawn="true"
    >
        <param name="sampling_freq" type="double" value="$(arg sampling_freq)" />
        <param name="pinger_freq" type="double" value="$(arg pinger_freq)" />
        <param name="noise_stdev" type="double" value="$(arg noise_stdev)" />
        <param name="x" type="double" value="$(arg x)" />
        <param name="y" type="double" value="$(arg y)" />
        <param name="rate" type="double" value="$(arg rate)" />
        <param name="data_port" type="string" value="tcp://127.0.0.1:6969" />
    </node>

    <!--sonar node-->
    <include file="$(find au_sonar)/launch/sonar.launch">
        <arg name="freq" value="$(arg pinger_freq)" />
        <arg name="data_port" value="tcp://127.0.0.1:6969" />
    </include>
</launch>
